usaspending_api/references/views_v2/autocomplete.py

```diff
@@ -106,6 +105,34 @@ def post(self, request):
         return self.agency_autocomplete_response(request)
 
 
+class NAICSAutocompleteViewSet(BaseAutocompleteViewSet):
+
+    def post(self, request):
+        """Return all budget function/subfunction titles matching the provided search text"""
+
+        search_text, limit = self.get_request_payload(request)
+
+        # get relevant TransactionContracts
+        queryset = TransactionContract.objects.filter(naics__isnull=False, naics_description__isnull=False)
+        # Filter based on search text
+        response = {}
+
+        queryset = queryset.annotate(similarity=Greatest(
+            TrigramSimilarity('naics', search_text),
+            TrigramSimilarity('naics_description', search_text)))\
+            .distinct().order_by('-similarity')
+
+        naics_exact_match_queryset = queryset.filter(similarity=1.0)
+        if naics_exact_match_queryset.count() > 0:
+            queryset = naics_exact_match_queryset
+
+        results_set = list(queryset.values('naics', 'naics_description')[:limit])
```

Comment:
If we're not longer relying on custom formatting or adding counts in the response, we could still leverage serializers here. Depends on the decision made about the counts in the response. The recipient autocomplete view set follows that example.